tensor.go

```diff
@@ -112,6 +100,22 @@ func (a Tensor) To(device Device, dtype int8) Tensor {
 	return Tensor{(*unsafe.Pointer)(&t)}
 }
 
+// CastTo cast tensor dtype
+func (a Tensor) CastTo(dtype int8) Tensor {
```

Comment:
Pytorch using `tensor.float()`, `tensor.int()` to cast to a type, do you want to follow this design pattern and provides `Float()`, `Int()` method to cast the Tensor?

Reply:
self.float() is equivalent to self.to(torch.float32).  Go does not support default parameters, `CastTo`is actually equal to `To(device=None)`. 

I will add `Float()` `Int()` if users needed.

Reply:
So do we need to provide `Float()` `Int()` method instead of `CastTo`, this is more similar to Pytorch.

Reply:
I believe that we need to expose the `Float` method too. I will add it in the next PR.